src/main/java/org/primefaces/component/datatable/export/DataTableExporter.java

```diff
@@ -337,6 +337,62 @@ public void export(FacesContext context, List<DataTable> tables, ExportConfigura
      */
     protected abstract void doExport(FacesContext facesContext, DataTable table, ExportConfiguration config, int index) throws IOException;
 
+    protected abstract String getContentType();
+
+    protected void sendExport2Client(String filenameWithExtension, ByteArrayOutputStream content, FacesContext context) throws IOException {
+        if (PrimeFaces.current().isAjaxRequest()) {
+            ajaxDownload(filenameWithExtension, content.toByteArray(), context);
+        }
+        else {
+            ExternalContext externalContext = context.getExternalContext();
+            externalContext.setResponseContentType(getContentType());
+            setResponseHeader(externalContext, ComponentUtils.createContentDisposition("attachment", filenameWithExtension));
+            externalContext.setResponseContentLength(content.size());
+            addResponseCookie(context);
+
+            content.writeTo(externalContext.getResponseOutputStream());
+            externalContext.responseFlushBuffer();
+        }
+    }
+
+    protected void sendExport2Client(String filenameWithExtension, StringBuilder content, String encodingType, FacesContext context) throws IOException {
+        if (PrimeFaces.current().isAjaxRequest()) {
+            ajaxDownload(filenameWithExtension, content.toString().getBytes(encodingType), encodingType, context);
+        }
+        else {
+            ExternalContext externalContext = context.getExternalContext();
+            externalContext.setResponseContentType(getContentType() + "; charset=" + encodingType);
+            setResponseHeader(externalContext, ComponentUtils.createContentDisposition("attachment", filenameWithExtension));
+
+            addResponseCookie(context);
+
+            OutputStream os = externalContext.getResponseOutputStream();
+            try (OutputStreamWriter osw = new OutputStreamWriter(os, encodingType);
+                 PrintWriter writer = new PrintWriter(osw);) {
+                writer.write(content.toString());
+                writer.flush();
+            }
+        }
+    }
+
+    private void ajaxDownload(String filenameWithExtension, byte[] content, FacesContext context) {
+        ajaxDownload(filenameWithExtension, content, null, context);
+    }
+
+    private void ajaxDownload(String filenameWithExtension, byte[] content, String encodingType, FacesContext context) {
+        String contentType = getContentType();
+        if (!LangUtils.isValueBlank(encodingType)) {
+            contentType += "; charset=" + encodingType;
+        }
+
+        String base64 = DatatypeConverter.printBase64Binary(content);
```

Comment:
This is the only line that gives me pause as it required JAXB to be on the classpath to work. What about `java.util.Base64` and doing
```java
String encoded = Base64.getEncoder().withoutPadding().encodeToString(someByteArray);
```

Reply:
Good Point. I also fixed this for DynamicContentSrcBuilder.

Reply:
Awesome! I think that is better